<h2> CREAZIONE LISTA 7352 COMUNI </h2>

<p> Ricavo dal Dataframe dei decessi totale dal 2015 al 30 giugno 2020 le lista dei comuni presenti che sono 7352. </p>

In [74]:
import pandas as pd
df = pd.read_csv('csv/decessi_istat_7352comuni.csv')
df.head()

REG NOME_REGIONE  PROV NOME_PROVINCIA  COD_PROVCOM NOME_COMUNE        DATA  \
0    1     Piemonte     1         Torino         1001       Agliè  2018-01-02   
1    1     Piemonte     1         Torino         1001       Agliè  2016-01-04   
2    1     Piemonte     1         Torino         1001       Agliè  2020-01-05   
3    1     Piemonte     1         Torino         1001       Agliè  2015-01-06   
4    1     Piemonte     1         Torino         1001       Agliè  2018-01-06   

   TOTALE  
0       1  
1       1  
2       1  
3       1  
4       2

<p> Prendo in esame il periodo fino al 30 aprile per poter fare il confronto nello stesso periodo nel DataFrame con 7898 comuni. </p> 

In [75]:
df['DATA'] = pd.to_datetime(df['DATA'])
df_data = df[df['DATA'] < '2020-05-01']

In [76]:
comuni = df_data['NOME_COMUNE'].unique()
comuni

array(['Agliè', 'Airasca', 'Ala di Stura', ..., 'Villasimius', 'Villasor',
       'Villaspeciosa'], dtype=object)

In [77]:
len(comuni)

7352

<p> La lista <b> comuni </b> contiene i 7352 comuni presenti del DataFrame dal 2015 al 30 giugno 2020. </p>

<h2> RICERCA DEI COMUNI MANCANTI </h2>

In [78]:
df2 = pd.read_csv('csv/decessi.csv', encoding = "ISO-8859-1")
df2.head()

REG  PROV NOME_REGIONE NOME_PROVINCIA NOME_COMUNE  COD_PROVCOM  \
0    1     1     Piemonte         Torino       Agliè         1001   
1    1     1     Piemonte         Torino       Agliè         1001   
2    1     1     Piemonte         Torino       Agliè         1001   
3    1     1     Piemonte         Torino       Agliè         1001   
4    1     1     Piemonte         Torino       Agliè         1001   

   TIPO_COMUNE  CL_ETA   GE        Data Maschi Femmine Totale  
0            1       7  609  09-06-2015      0       0      0  
1            1       7  609  09-06-2016      0       0      0  
2            1       7  609  09-06-2017      0       0      0  
3            1       7  609  09-06-2018      0       0      0  
4            1       7  609  09-06-2019      1       0      1

In [79]:
#Numero totale di comuni 
len(df2['NOME_COMUNE'].unique())

7898

In [80]:
#Numero totale delle province
len(df2['NOME_PROVINCIA'].unique())

107

In [81]:
comuni_mancanti = df2[df2['NOME_COMUNE'].isin(comuni) == False]
comuni_mancanti.head()

REG  PROV NOME_REGIONE NOME_PROVINCIA NOME_COMUNE  COD_PROVCOM  \
17250    1     1     Piemonte         Torino   Balangero         1016   
17251    1     1     Piemonte         Torino   Balangero         1016   
17252    1     1     Piemonte         Torino   Balangero         1016   
17253    1     1     Piemonte         Torino   Balangero         1016   
17254    1     1     Piemonte         Torino   Balangero         1016   

       TIPO_COMUNE  CL_ETA   GE        Data Maschi Femmine Totale  
17250            2       6  821  21-08-2015      0       0      0  
17251            2       6  821  21-08-2016      0       0      0  
17252            2       6  821  21-08-2017      1       0      1  
17253            2       6  821  21-08-2018      0       0      0  
17254            2       6  821  21-08-2019      0       0      0

In [82]:
len(comuni_mancanti['NOME_COMUNE'].unique())

546

<p> Il DataFrame <b> comuni_mancanti </b> contiene tutti i comuni che non sono presenti nel DataFrame usato per l'analisi dei decessi.
    <br />
    Nello specifico, si tratta di <b>546</b> comuni.
</p>

<h3> Modifica del DataFrame comuni_mancanti </h3>

In [83]:
#elimino le colonne che al momento non ci interessano
comuni_mancanti.drop(['REG', 'PROV', 'NOME_REGIONE', 'COD_PROVCOM', 'TIPO_COMUNE',
                 'CL_ETA', 'GE', 'Maschi', 'Femmine', ' Data', 'Maschi', 'Femmine', 'Totale'], axis=1, inplace=True)
comuni_mancanti.head()

c:\users\monta\appdata\local\programs\python\python38\lib\site-packages\pandas\core\frame.py:3990: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


NOME_PROVINCIA NOME_COMUNE
17250         Torino   Balangero
17251         Torino   Balangero
17252         Torino   Balangero
17253         Torino   Balangero
17254         Torino   Balangero

In [84]:
# salvo il DataFrame
comuni_mancanti.to_csv('csv/comuni_province_mancanti.csv', index=False)

<h3> RICERCA DELLE PROVINCE DI CUI NON ABBIAMO I DATI DI TUTTI I COMUNI CHE NE FANNO PARTE </h3>

In [85]:
lista_province = comuni_mancanti['NOME_PROVINCIA'].unique()
lista_province

array(['Torino', 'Vercelli', 'Novara', 'Cuneo', 'Asti', 'Alessandria',
       "Valle d'Aosta/Vallée d'Aoste", 'Imperia', 'Savona', 'Genova',
       'La Spezia', 'Varese', 'Como', 'Milano', 'Bergamo', 'Brescia',
       'Pavia', 'Cremona', 'Mantova', 'Bolzano/Bozen', 'Trento', 'Verona',
       'Vicenza', 'Belluno', 'Treviso', 'Venezia', 'Padova', 'Udine',
       'Parma', "Reggio nell'Emilia", 'Modena', 'Bologna', 'Ferrara',
       'Forlì-Cesena', 'Pesaro e Urbino', 'Ancona', 'Macerata',
       'Ascoli Piceno', 'Massa-Carrara', 'Lucca', 'Pistoia', 'Firenze',
       'Livorno', 'Pisa', 'Arezzo', 'Siena', 'Grosseto', 'Perugia',
       'Terni', 'Viterbo', 'Rieti', 'Roma', 'Latina', 'Frosinone',
       'Caserta', 'Benevento', 'Napoli', 'Avellino', 'Salerno',
       "L'Aquila", 'Teramo', 'Pescara', 'Chieti', 'Campobasso', 'Foggia',
       'Bari', 'Taranto', 'Brindisi', 'Lecce', 'Potenza', 'Matera',
       'Cosenza', 'Catanzaro', 'Reggio Calabria', 'Trapani', 'Palermo',
       'Messina', 'Agrige

In [86]:
len(lista_province)

99

<p>La lista <b>lista_province</b> contiente il nome di tutte le province di cui non abbiamo i dati completi per ogni comune. Le province in questione risultano essere <b>99</b>. </p>

<h2> ANALISI DEI DATI RICAVATI DALLA PROTEZIONE CIVILE SU BASE PROVINCIALE </h2>

In [87]:
import pandas as pd 
df_covid = pd.read_csv('csv/covid19_province_ita.csv')
df_covid.head()

data stato  codice_regione denominazione_regione  \
0  2020-02-24T18:00:00   ITA              13               Abruzzo   
1  2020-02-24T18:00:00   ITA              13               Abruzzo   
2  2020-02-24T18:00:00   ITA              13               Abruzzo   
3  2020-02-24T18:00:00   ITA              13               Abruzzo   
4  2020-02-24T18:00:00   ITA              13               Abruzzo   

   codice_provincia               denominazione_provincia sigla_provincia  \
0                66                              L'Aquila              AQ   
1                67                                Teramo              TE   
2                68                               Pescara              PE   
3                69                                Chieti              CH   
4               979  In fase di definizione/aggiornamento             NaN   

         lat       long  totale_casi note  
0  42.351222  13.398438            0  NaN  
1  42.658918  13.704400            0  NaN  
2  42.464584  14.213648            0  NaN  
3  42.351032  14.167546            0  NaN  
4        NaN        NaN            0  NaN

In [88]:
# elimino dalle province le voci
df_covid = df_covid.set_index('denominazione_provincia')
df_covid = df_covid.drop('Fuori Regione / Provincia Autonoma', axis=0)
df_covid = df_covid.drop('In fase di definizione/aggiornamento', axis=0)
df_covid = df_covid.reset_index()

In [89]:
# modifico il formato delle date
df_covid['data'] = pd.to_datetime(df_covid['data'], format='%Y-%m-%d')
# elimino l'orario dal campo data
df_covid['data'] = df_covid['data'].dt.strftime('%Y-%m-%d')
# modifico il formato dei casi
df_covid['totale_casi'] = pd.to_numeric(df_covid['totale_casi'])

In [90]:
#prendo i valori fino al 30 giugno in quanto la ts dei decessi dei 7352 comuni termina il 30 giugno
df_covid = df_covid[df_covid['data'] < '2020-07-01']

In [91]:
#elimino le colonne che al momento non ci interessano
df_covid.drop(['stato', 'codice_regione', 'codice_provincia', 'sigla_provincia', 'lat', 'long', 'note' ], axis=1, inplace=True)
df_covid.head()

denominazione_provincia        data denominazione_regione  totale_casi
0                L'Aquila  2020-02-24               Abruzzo            0
1                  Teramo  2020-02-24               Abruzzo            0
2                 Pescara  2020-02-24               Abruzzo            0
3                  Chieti  2020-02-24               Abruzzo            0
4                 Potenza  2020-02-24            Basilicata            0

<h2> RICERCA DELLE PROVINCE COMPLETE DI TUTTI I COMUNI </h2>

In [92]:
#Numero delle province presenti nei dati della Protezione Civile
len(df_covid['denominazione_provincia'].unique())

107

In [93]:
province_complete = df_covid[df_covid['denominazione_provincia'].isin(lista_province) == False]
province_complete

denominazione_provincia        data  denominazione_regione  totale_casi
8          Reggio di Calabria  2020-02-24               Calabria            0
16                   Piacenza  2020-02-24         Emilia-Romagna            0
22                    Ravenna  2020-02-24         Emilia-Romagna            0
26                    Gorizia  2020-02-24  Friuli Venezia Giulia            0
27                    Trieste  2020-02-24  Friuli Venezia Giulia            0
...                       ...         ...                    ...          ...
13646                 Bolzano  2020-06-30           P.A. Bolzano         2639
13674                  Ragusa  2020-06-30                Sicilia           87
13676           Massa Carrara  2020-06-30                Toscana         1051
13688                   Aosta  2020-06-30          Valle d'Aosta         1195
13695                  Rovigo  2020-06-30                 Veneto          444

[1536 rows x 4 columns]

<p> Il DataFrame <b>province_complete</b> contiene i dati realativi alle province di cui possediamo i dati per ogni comune che le compone </p>

In [94]:
#Ricerca del nome delle province complete
province = province_complete['denominazione_provincia'].unique()

In [95]:
province

array(['Reggio di Calabria', 'Piacenza', 'Ravenna', 'Gorizia', 'Trieste',
       'Pordenone', 'Sondrio', 'Bolzano', 'Ragusa', 'Massa Carrara',
       'Aosta', 'Rovigo'], dtype=object)

In [96]:
#Elimino le province che risultano essere scritte in maniera diversa 
province_complete = province_complete.set_index('denominazione_provincia')
province_complete = province_complete.drop('Massa Carrara', axis=0)
province_complete = province_complete.drop('Reggio di Calabria', axis=0)
province_complete = province_complete.drop('Aosta', axis=0)
province_complete = province_complete.drop('Bolzano', axis=0)
province_complete = province_complete.reset_index()

In [97]:
# salvo il DataFrame con i dati delle province complete relativi ai decessi covid da Protezione Civile
province_complete.to_csv('csv/province_complete.csv', index=False)

In [98]:
province = province_complete['denominazione_provincia'].unique()

In [99]:
#Lista delle province di cui abbiamo i dati per ogni comune
province

array(['Piacenza', 'Ravenna', 'Gorizia', 'Trieste', 'Pordenone',
       'Sondrio', 'Ragusa', 'Rovigo'], dtype=object)

In [100]:
len(province)

8

<h2> CONCLUSIONE </h2>

<p> Le province di cui non possediamo i dati di tutti i comuni sono <b>99</b>. Per cui, andandole a eliminare dai dati della Protezione Civile che comprendono <b> 107 </b> province, risulta che le province per cui abbiamo i dati completi per ogni comune appartenente sono <b> 8 </b>.
    </p>

<h1> CREAZIONE DELLA NUOVA SERIE STORICA </h1>

<p> Nella nuova serie storica sono presenti i decessi dal 2015 al 30 giugno 2020 dei comuni che fanno parte della provincia di <b>Piacenza</b>, <b>Ravenna</b>, <b>Gorizia</b>, <b>Trieste</b>, <b>Pordenone</b>, <b>Sondrio</b>, <b>Ragusa</b> e <b>Rovigo</b>.

In [101]:
df_comuni_completi = df[df['NOME_PROVINCIA'].isin(province)]
df_comuni_completi

REG           NOME_REGIONE  PROV NOME_PROVINCIA  COD_PROVCOM  \
282858     3              Lombardia    14        Sondrio        14001   
282859     3              Lombardia    14        Sondrio        14001   
282860     3              Lombardia    14        Sondrio        14001   
282861     3              Lombardia    14        Sondrio        14001   
282862     3              Lombardia    14        Sondrio        14001   
...      ...                    ...   ...            ...          ...   
1889755    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889756    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889757    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889758    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889759    6  Friuli-Venezia Giulia    93      Pordenone        93053   

                    NOME_COMUNE       DATA  TOTALE  
282858   Albaredo per San Marco 2018-01-15       1  
282859   Albaredo per San Marco 2018-01-20       1  
282860   Albaredo per San Marco 2020-01-29       1  
282861   Albaredo per San Marco 2015-01-30       1  
282862   Albaredo per San Marco 2017-02-15       1  
...                         ...        ...     ...  
1889755        Valvasone Arzene 2017-12-24       1  
1889756        Valvasone Arzene 2015-12-27       1  
1889757        Valvasone Arzene 2017-12-27       1  
1889758        Valvasone Arzene 2018-12-29       1  
1889759        Valvasone Arzene 2019-12-31       1  

[87108 rows x 8 columns]

In [102]:
# salvo il DataFrame
df_comuni_completi.to_csv('csv/df_comuni_completi.csv', index=False)